install/project/includes/configuration/configuration.inc.sample.php

```diff
@@ -165,9 +166,11 @@
 			define ('__APP_CSS_ASSETS__', __PROJECT_ASSETS__ . '/css');
 			define ('__APP_IMAGE_ASSETS__', __PROJECT_ASSETS__ . '/images');
 			define ('__APP_PHP_ASSETS__', __PROJECT_ASSETS__ . '/php');
+			define ('__APP_IMAGE_CACHE__', __APP_IMAGE_ASSETS__ . '/cache');
+			define ('__APP_UPLOAD__', __DOCROOT__ . __PROJECT_ASSETS__ . '/upload');
```

Comment:
As @kukrik mentioned already, we need two cache folders: one for php-accessible things, and it should be full filesystem path with `__DOCROOT__`, and another one for web-accesible goods, and it should be relative to site domen, e.g. for `http://example.com/project/assets/cache/myphoto.png` it should be `/project/assets/cache`

For the first one we have `__TMP__`. For the second we don't have one place, but only images cache in the `__APP_IMAGE_CACHE__` define. We should have `__APP_CACHE_ASSETS__ `instead and, may be, `__APP_IMAGE_CACHE_ASSETS__` defined as `__APP_CACHE_ASSETS__  . '/images'`. And we need to have `__APP_CACHE__` as a full-path version of the  `__APP_CACHE_ASSETS__` with a `__DOCROOT__` included. The `__APP_IMAGE_CACHE__` becomes a full-path version of the `__APP_IMAGE_CACHE_ASSETS__` define.

Uploads are different. All other caches a volatile and can be cleared in any time. Uploads are not. They are a part of user expirience and should be preserved. That is a reason to maintain it separately from cache directory. It means that the current `__APP_UPLOAD__` is done right. But it lacks it's `ASSETS` counterpart: `__APP_UPLOAD_ASSETS__` that is needed to be used by client code to form URI-s of uploaded items. The original `__APP_UPLOAD__` is used by the code that needs to work with file in a filesystem.

If it is accepted, I can implement these changes.